cli/src/cli/run.rs

```diff
@@ -11,95 +11,181 @@ use holochain_conductor_api::{
+        interface_type, port
+    );
+    println!("Type 'exit' to stop the conductor and exit the program");
+
+    let mut rl = rustyline::Editor::<()>::new();
+
+    loop {
+        let readline = rl.readline("hc> ")?;
+
+        match readline.as_str().trim() {
+            "exit" => break,
+            other if !other.is_empty() => eprintln!(
+                "command {} not recognized. Available commands are: exit",
+                other.red().bold()
+            ),
+            _ => continue,
+        }
     }
 
+    Ok(())
+}
+
+pub fn get_interface_type_string(given_type: String) -> String {
+    // note that this behaviour is documented within
+    // holochain_common::env_vars module and should be updated
+    // if this logic changes
+    // The environment variable overrides the CLI flag
+    EnvVar::Interface.value().ok().unwrap_or_else(|| given_type)
+}
+
+pub fn hc_run_configuration(
+    dna_path: &PathBuf,
+    port: u16,
+    persist: bool,
+    networked: bool,
+    interface_type: &String,
+) -> DefaultResult<Configuration> {
+    Ok(Configuration {
+        agents: vec![agent_configuration()],
+        dnas: vec![dna_configuration(&dna_path)],
+        instances: vec![instance_configuration(storage_configuration(persist)?)],
+        interfaces: vec![interface_configuration(&interface_type, port)?],
+        network: networking_configuration(networked),
+        logger: logger_configuration(),
+        ..Default::default()
+    })
+}
+
+// AGENT
+const AGENT_NAME_DEFAULT: &str = "testAgent";
+const AGENT_CONFIG_ID: &str = "hc-run-agent";
+
+fn agent_configuration() -> AgentConfiguration {
     // note that this behaviour is documented within
     // holochain_common::env_vars module and should be updated
     // if this logic changes
     let agent_name = EnvVar::Agent
         .value()
         .ok()
-        .unwrap_or_else(|| String::from("testAgent"));
+        .unwrap_or_else(|| String::from(AGENT_NAME_DEFAULT));
     let keypair = test_key(&agent_name);
     let pub_key = KeyBuffer::with_corrected(&keypair.get_id()).unwrap();
     let agent_id = AgentId::new(&agent_name, &pub_key);
-    let agent_config = AgentConfiguration {
+    AgentConfiguration {
         id: AGENT_CONFIG_ID.into(),
         name: agent_id.nick,
         public_address: agent_id.key,
         key_file: agent_name,
         holo_remote_key: None,
-    };
+    }
+}
+
+// DNA
+const DNA_CONFIG_ID: &str = "hc-run-dna";
 
-    let dna_config = DnaConfiguration {
+fn dna_configuration(dna_path: &PathBuf) -> DnaConfiguration {
+    DnaConfiguration {
         id: DNA_CONFIG_ID.into(),
-        file: crate::util::std_dna_file_name(path)?,
+        file: dna_path
+            .to_str()
+            .expect("Expected DNA path to be valid unicode")
+            .to_string(),
         hash: None,
-    };
+    }
+}
 
-    let storage = if persist {
+// STORAGE
+const LOCAL_STORAGE_PATH: &str = ".hc";
```

Comment:
Isn't this something that should either go to `holochain_common` or just use the path from there?

Reply:
I think if anything should change, it’s the folder name itself... it’s a bit different than the others because this directory sits inside the NDA folder itself